tool_quality_checks/importer.py

```diff
@@ -29,34 +94,34 @@ def copy2pg_database(settings, in_filepath, layer_name, schema="public"):
         "OVERWRITE=YES",
         "SCHEMA={}".format(schema),
         "SPATIAL_INDEX=GIST",
+        # try to comment row below when running to transaction block error
         "FID=id",
         "PRECISION=NO",
         "GEOMETRY_NAME=geom",
         "DIM=2",
     ]
-    try:
-        ogr.RegisterAll()
-        # TODO srid is now based on in_layer, which could be a strange spatial reference
-        # TODO findout how to make the target ref 28992 by default
-        new_layer = datasource.CreateLayer(
-            layer_name, in_layer.GetSpatialRef(), in_layer.GetGeomType(), options
-        )
-        for x in range(in_layer.GetLayerDefn().GetFieldCount()):
-            new_layer.CreateField(in_layer.GetLayerDefn().GetFieldDefn(x))
-
-        new_layer.StartTransaction()
-        for x in range(in_layer.GetFeatureCount()):
-            new_feature = in_layer.GetFeature(x)
-            new_feature.SetFID(-1)
-            new_layer.CreateFeature(new_feature)
-            if x % 128 == 0:
-                new_layer.CommitTransaction()
-                new_layer.StartTransaction()
-        new_layer.CommitTransaction()
-
-    except Exception as e:
-        logger.warning(e)
-        raise
+
+    ogr.RegisterAll()
+    # TODO srid is now based on in_layer, which could be a strange spatial reference
+    # TODO findout how to make the target ref 28992 by default
+    new_layer = datasource.CreateLayer(
+        correct_layer_name,
+        correct_in_layer.GetSpatialRef(),
+        correct_in_layer.GetGeomType(),
+        options,
+    )
+    for x in range(correct_in_layer.GetLayerDefn().GetFieldCount()):
+        new_layer.CreateField(correct_in_layer.GetLayerDefn().GetFieldDefn(x))
+
+    new_layer.StartTransaction()
+    for x in range(correct_in_layer.GetFeatureCount()):
```

Comment:
Niet twee keer dezelfde tijdelijke variabele gebruiken in dezelfde functie. Elk jaar vind ik bij mezelf wel eens een bug waar ik dit gedaan heb.

Conventie 2: de eerste keer bij zo'n loop dus `i`, de tweede keer `j`.